courseraresearchexports/exports/utils.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Coursera
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from courseraresearchexports.exports.constants import COURSE_API, PARTNER_API,\
+    ANONYMITY_LEVELS, EXPORT_TYPES, SCHEMA_NAMES
+from urlparse import urlparse
+from tqdm import tqdm
+import requests
+import os
+import logging
+import zipfile
+
+
+def extract_export_archive(export_archive, dest, delete_archive=True):
+    """
+    Extracts a compressed folder
+    :param export_archive:
+    :param dest:
+    :param delete_archive: delete the archive after extracting
+    :return:
+    """
+    try:
+        with zipfile.ZipFile(export_archive, 'r') as z:
+            z.extractall(dest)
+        if delete_archive:
+            os.remove(export_archive)
+        return dest
+    except:
+        logging.error('Error in extracting export archive {} to {}'.format(
+            export_archive, dest))
+        raise
+
+
+def download_export(export_job, dest):
+    """
+    Download an export archive if ready given an export_job_id
+    :param export_job:
+    :param dest:
+    :return:
+    """
+    if export_job['exportType'] == 'RESEARCH_WITH_EVENTING':
+        logging.error('Generate download links to download eventing exports')
+        raise ValueError('Require RESEARCH_WITH_SCHEMAS job')
+    elif 'downloadLink' not in export_job:
+        logging.error('Export job {} is not ready'.format(export_job['id']))
+        raise ValueError('Export job is not yet ready for download')
+
+    if not os.path.exists(dest):
+        os.makedirs(dest)
+    download_link = export_job['downloadLink']
+    export_filename = urlparse(download_link).path.split('/')[-1]
+    response = requests.get(download_link, stream=True)
+    chunk_size = 1024 * 1024
+    output_filename = os.path.join(dest, export_filename)
+
+    with open(output_filename, 'wb') as f:
+        for data in tqdm(
+                iterable=response.iter_content(chunk_size),
+                total=int(response.headers['Content-length'])/chunk_size,
+                unit='MB',
+                desc=export_filename):
+
+            f.write(data)
+
+    return output_filename
+
+
+def lookup_course_slug_by_id(course_id):
```

Comment:
It's for display purposes since the `onDemandExports` only returns the course/partner id.  I can go either way, I'm only using it during the `getAll` command so the table is more readable.